lib/config.js

```diff
@@ -69,7 +72,53 @@ const getConfig = async (args) => {
     console.log('Error:', error.message);
   }
 
-  return mergeOptions(defaultOptions, contentfulCliOptions, environmentOptions, configFileOptions, args || {});
+  let userConfigFileOptions = {};
+  if (args.config) {
+    try {
+      let config = {};
+      let filepath = '';
+      if (args.config.endsWith('.js') || args.config.endsWith('.ts')) {
+        config = await jiti.import(path.resolve(args.config));
+        filepath = jiti.esmResolve(path.resolve(args.config));
+      } else {
+        const explorer = cosmiconfig('migrations');
+        const explorerResult = await explorer.load(args.config);
+        config = explorerResult.config;
+        filepath = explorerResult.filepath;
+      }
+      if (config) {
+        userConfigFileOptions = {
+          directory: path.resolve(path.dirname(filepath || ''), args.directory || 'migrations'),
+          ...(config || {}),
+        };
+
+        if (userConfigFileOptions.directory && !path.isAbsolute(userConfigFileOptions.directory)) {
+          userConfigFileOptions.directory = path.resolve(path.dirname(filepath || ''), userConfigFileOptions.directory);
+        }
```

Comment:
This whole getConfig function got quite long and is hard to digest.
It seems this block is the same logic as above where configFileOptions is created. I think it could help to extract this code into a little helper function.